standalone/TunnelingPath.cpp

```diff
@@ -80,7 +80,7 @@ int main()
     BounceActionInt *ClosestBounceActionInt;
     std::cout << "Found a transitions with Tp =\t"
               << bounce.GetPercolationTemp() << " GeV.\n";
-    for (auto BAInt : bounce.SolutionList)
+    for (auto &BAInt : bounce.SolutionList)
```

Comment:
```suggestion
    for (const auto &BAInt : bounce.SolutionList)
```

Your problem is not with the c++17 change. Before you made a copy of this object. If this has a pointer then you make a shallow copy which does not receive updates when it changes from a different thread.

I don't know if it compiles here with const ref, but maybe try it.

Reply:
It does not compile.

```bash
[ 98%] Built target GenericTests
/Users/joaofranciscoviana/mega/BSMPT/standalone/TunnelingPath.cpp:88:34: error: assigning to 'BSMPT::BounceActionInt *' from 'BSMPT::BounceActionInt const *' discards qualifiers
        ClosestBounceActionInt = &BAInt;
                                 ^~~~~~
1 error generated.
```

 I mentioned the c++14 because I was getting the correct output at some point and I don't remember changing anything else.

Reply:
Fixed. Had to convert some member functions to const to make it work.